store/sqlstore/compliance_store.go

```diff
@@ -36,19 +37,19 @@ func newSqlComplianceStore(sqlStore SqlStore) store.ComplianceStore {
 func (s SqlComplianceStore) createIndexesIfNotExists() {
 }
 
-func (s SqlComplianceStore) Save(compliance *model.Compliance) (*model.Compliance, *model.AppError) {
+func (s SqlComplianceStore) Save(compliance *model.Compliance) (*model.Compliance, error) {
 	compliance.PreSave()
 	if err := compliance.IsValid(); err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "error validating compliance")
```

Comment:
You shouldn't be wrapping this error, because we need the real underneath data to handle properly this case (this will change in the future, but for now, just `return nil, err`)